src/transformers/utils/logging.py

```diff
@@ -30,6 +31,50 @@
 _lock = threading.Lock()
 _default_handler: Optional[logging.Handler] = None
 
+log_levels = {
+    "debug": logging.DEBUG,
+    "info": logging.INFO,
+    "warning": logging.WARNING,
+    "error": logging.ERROR,
+    "critical": logging.CRITICAL,
+}
+
+_default_log_level = logging.WARNING
+
+
+def _get_default_logging_level():
+    """
+    If TRANSFORMERS_VERBOSITY env var is set to one of the valid choices return that as the new default level.
+    If it is not - fall back to ``_default_log_level``
+    """
+    env_level_str = os.getenv("TRANSFORMERS_VERBOSITY", None)
```

Comment:
Shouldn't `logging_level_str_to_code ` be used to parse the env var here?

Reply:
You read my mind. As you can see I re-coded it to not be fatal if someone mistyped the verbosity string in the env var. But yes, it could totally be re-used if we make it a fatal error. Thoughts?

Reply:
Since we need to be specific about this being a failed attempt at env var setting, if we go for not ignoring an invalid env var with just logging a warning, then it gets recoded it to indicate that it's env var that's failing:

```
def _get_default_logging_level():
    env_level_str = os.getenv("TRANSFORMERS_VERBOSITY", None)
    if env_level_str:
    try:
         return logging_level_str_to_code(level_str):
    except:
            logging.getLogger().warning(
                f"Unknown option TRANSFORMERS_VERBOSITY={env_level_str}, "
                f"has to be one of: { ', '.join(log_levels.keys()) }"
            )
    return _default_log_level
```

Reply:
Okay that makes sense, we can keep it the way it is.